services/productService.js

```diff
@@ -2,8 +2,26 @@ const productDao = require('../models/productDao');
 
 const getProducts = async () => {
   return await productDao.getProducts();
-};
+}
+
+const getProductMainCategories = async (mainCategoryName, sort) => {
+  if (!sort) {
+    return await productDao.getProductMainCategories(mainCategoryName);
+  } else {
+    return await productDao.mainCategoryFiltering(mainCategoryName, sort);
+  }
+}
```

Comment:
삼항연산자 사용하는 방법도 있으니 참고해주세요!
```suggestion
    return sort
    ? await productDao.mainCategoryFiltering(mainCategoryName, sort);
    : await productDao.getProductMainCategories(mainCategoryName);
 
```

Reply:
넵!! 날짜 필터링 기능도 구현해서 if문 사용하였습니다.